release.sh

```diff
@@ -4,17 +4,55 @@
 # we support with the golang cross-compiler.
 #
 # Copyright (c) 2016 Company 0, LLC.
-# Copyright (c) 2019 Lightning Labs
+# Copyright (c) 2020 Lightning Labs
 # Use of this source code is governed by the ISC
 # license.
 
+# Exit on errors.
+set -e
+
 # If no tag specified, use date + version otherwise use tag.
 if [[ $1x = x ]]; then
     DATE=`date +%Y%m%d`
     VERSION="01"
     TAG=$DATE-$VERSION
 else
     TAG=$1
+
+    # If a tag is specified, ensure that that tag is present and checked out.
+    if [[ $TAG != $(git describe) ]]; then
+        echo "tag $TAG not checked out"
+        exit 1
+    fi
+
+    # Verify that it is signed.
+    if ! git verify-tag $TAG; then 
+        echo "tag $TAG not signed"
+        exit 1
+    fi
+
+    # Build loop to extract version.
+    make build
```

Comment:
i wonder if it's a good idea to also make the `build` target the default in our make file. Kind of weird that we need to always do a `make build`

Reply:
added default target